### PR TITLE
33 create profile menu

### PR DESCRIPTION
**Pull request by muratcan23**

### CODE REVIEW

File: src/pages/Home/Components/ProfileMenu.tsx [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+                <Text fontSize="small" ml="2">
+                  Settings
+                </Text>
+              </Box>
+            </Flex>
+          </Box>
+        </Flex>
+
+        <Divider />
+        <MenuItem
```
Comment (kod-man) [inline src/pages/Home/Components/ProfileMenu.tsx:87]: Why don't you create a component for MenuItem? You are repeating yourself. You should obey the Don't Repeat Yourself (DRY) rule.

**Review by kod-man — CHANGES_REQUESTED** [review]

File: src/pages/Home/Components/ProfileMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+                  </Text>
+                  <Text fontSize="xm">muratal0606@gmail.com</Text>
+                </Flex>
+              </VStack>
+            </Flex>
+            <Flex alignItems="center" ml={2} mb="1">
+              <Box h={4} w={4} mr="1" justifyContent="flex-start">
+                <FiSettings color="gray" />
+              </Box>
+              <Box>
```
Comment (kod-man) [inline src/pages/Home/Components/ProfileMenu.tsx:88]: Box is not needed

File: src/pages/Home/Components/ProfileMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+          </Box>
+        </Flex>
+        <Divider />
+
+        {dataProfileMenu.map((item) => (
+          <ProfileCards key={item.text} text={item.text} Icon={item.Icon} />
+        ))}
+
+        <Divider />
+        <MenuItem
```
Comment (kod-man) [inline src/pages/Home/Components/ProfileMenu.tsx:103]: You can use Profile card components once again for log out

Comment (kod-man) [reply src/pages/Home/Components/ProfileMenu.tsx:103]: Because the only difference is the divider

**Review by kod-man — CHANGES_REQUESTED** [review]

File: src/pages/Home/Components/ProfileMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+              >
+                MC
+              </Flex>
+              <VStack>
+                <Flex flexDir="column" m={2}>
+                  <Text as="b" fontSize="sm">
+                    Murat Can
+                  </Text>
+                  <Text fontSize="xm">muratal0606@gmail.com</Text>
+                </Flex>
```
[the remark this answers is not in the document]
Comment (muratcan23) [reply src/pages/Home/Components/ProfileMenu.tsx:81]: i feel ashamed :(

File: src/pages/Home/Components/ProfileCards.tsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
+import { Box, Flex, MenuItem, Text } from "@chakra-ui/react";
+import { FC } from "react";
+import { IconType } from "react-icons";
+
+type CardProps = {
+  Icon: IconType;
+  text: string;
+};
+
+const ProfileCards: FC<CardProps> = ({ Icon, text }) => {
```
[the remark this answers is not in the document]
Comment (muratcan23) [reply src/pages/Home/Components/ProfileCards.tsx:10]: this will push me a little bit.

File: src/pages/Home/Components/ProfileMenu.tsx [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+          </Box>
+        </Flex>
+        <Divider />
+
+        {dataProfileMenu.map((item) => (
+          <ProfileCards key={item.text} text={item.text} Icon={item.Icon} />
+        ))}
+
+        <Divider />
+        <ProfileCards text={"Logout"} Icon={FiSettings} />
```
Comment (kod-man) [inline src/pages/Home/Components/ProfileMenu.tsx:100]: no need to add curly braces {}

File: src/pages/Home/Components/ProfileMenu.tsx [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+                borderRadius="50%"
+              >
+                MC
+              </Flex>
+              <VStack>
+                <Flex flexDir="column" m={2}>
+                  <Text as="b" fontSize="sm">
+                    Murat Can
+                  </Text>
+                  <Text fontSize="xs" color="grey">
```
Comment (kod-man) [inline src/pages/Home/Components/ProfileMenu.tsx:77]: grey ? should not it be gray?

File: src/pages/Home/Components/ProfileCards.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (kod-man) [reply src/pages/Home/Components/ProfileCards.tsx:10]: Screenshot 2022-12-25 at 13 17 11

[the remark this answers is not in the document]
Comment (kod-man) [reply src/pages/Home/Components/ProfileCards.tsx:10]: try this one

**Review by kod-man — CHANGES_REQUESTED** [review]

**Review by kod-man — APPROVED** [review]

**Pull request merged**